src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/span_links/index.tsx

```diff
@@ -0,0 +1,230 @@
+
+  useEffect(() => {
+    const hasData = value.incomingSpanLinks.length > 0 || value.outgoingSpanLinks.length > 0;
+    if (hasData) {
+      if (type === 'incoming' && value.incomingSpanLinks.length === 0) {
+        setType('outgoing');
+      } else if (type === 'outgoing' && value.outgoingSpanLinks.length === 0) {
+        setType('incoming');
+      }
+    }
+  }, [value, type]);
+
+  const selectOptions: EuiSelectOption[] = useMemo(
+    () => [
+      {
+        'data-test-subj': 'unifiedDocViewerSpanLinkTypeSelect-incoming',
+        value: 'incoming',
+        text: i18n.translate(
+          'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.combo.incomingLinks',
+          {
+            defaultMessage: 'Incoming links ({incomingLinks})',
+            values: { incomingLinks: value.incomingSpanLinks.length },
+          }
+        ),
+        disabled: !value.incomingSpanLinks.length,
+      },
+      {
+        'data-test-subj': 'unifiedDocViewerSpanLinkTypeSelect-outgoing',
+        value: 'outgoing',
+        text: i18n.translate(
+          'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.combo.outgoingLinks',
+          {
+            defaultMessage: 'Outgoing links ({outgoingLinks})',
+            values: { outgoingLinks: value.outgoingSpanLinks.length },
+          }
+        ),
+        disabled: !value.outgoingSpanLinks.length,
+      },
+    ],
+    [value]
+  );
+
+  const columns = useMemo(
+    () => getColumns({ generateDiscoverLink, type }),
+    [generateDiscoverLink, type]
+  );
+
+  const openInDiscoverLink = useMemo(() => {
+    if (type === 'incoming') {
+      return generateDiscoverLink(getIncomingSpanLinksESQL(traceId, docId));
+    }
+
+    return generateDiscoverLink(getOutgoingSpanLinksESQL(spanLinks));
+  }, [docId, generateDiscoverLink, spanLinks, traceId, type]);
+
+  if (
+    loading ||
+    (!error && value.incomingSpanLinks.length === 0 && value.outgoingSpanLinks.length === 0)
+  ) {
+    return null;
+  }
+
+  return (
+    <ContentFrameworkSection
+      data-test-subj="unifiedDocViewerSpanLinksAccordion"
+      id="spanLinksSection"
+      title={i18n.translate(
+        'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks',
+        { defaultMessage: 'Span links' }
+      )}
+      actions={
+        openInDiscoverLink
+          ? [
+              {
+                icon: 'discoverApp',
+                label: i18n.translate(
+                  'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.openInDiscover',
+                  { defaultMessage: 'Open in discover' }
+                ),
+                ariaLabel: i18n.translate(
+                  'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.openInDiscover',
+                  { defaultMessage: 'Open in discover link' }
+                ),
+                href: openInDiscoverLink,
```

Comment:
Now the link is not working 🤔

Reply:
All my links are working properly, could you try it again, please?